### PR TITLE
Add Logistic distribution

### PR DESCRIPTION
**Pull request by DenisKolodin**
A logistic distribution implementation. I tried to take into account all the previous comments.
It seems all the methods suitable for inlining.

### CODE REVIEW

File: src/distribution/logistic.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    /// Return the scale parameter.
+    #[inline(always)]
+    pub fn s(&self) -> f64 { self.s }
+}
+
+impl Default for Logistic {
+    #[inline]
+    fn default() -> Self {
+        Logistic::new(0.0, 1.0)
```
Comment (IvanUkhov) [inline src/distribution/logistic.rs]: I’m not that familiar with the logistic distribution. Is there such thing as the standard logistic distribution (analogous to the standard Gaussian distribution, for instance)?

File: src/distribution/logistic.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    fn entropy() {
+        assert_eq!(new!(0.0, (-2.0f64).exp()).entropy(), 0.0);
+    }
+
+    #[test]
+    fn inverse() {
+        let d = new!(5.0, 5.0);
+        let p = vec![0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0];
+        let x = vec![
+                      NEG_INFINITY, -5.986122886681094e+00, -1.931471805599453e+00,
```
Comment (IvanUkhov) [inline src/distribution/logistic.rs]: If you want the numbers to align, please align them with respect to the decimal point; don’t eliminate digits after the decimal point just to accommodate the minus sign.

**Comment by IvanUkhov** [comment]
Great work! My only question is if `Default` should be implemented for `Logistic`. It kind of makes sense to implement `Default` for `Exponential`, `Uniform`, and `Gaussian`, but I don’t know much about `Logistic`. So it’s up to you.

File: src/distribution/logistic.rs (hunk 1 above)
Comment (DenisKolodin) [inline src/distribution/logistic.rs]: It's bell-curved too and really close to gaussian, but different. Wiki have good explanation about it: 

> Indeed, the logistic and normal distributions have a quite similar shape. However, the logistic distribution has heavier tails, which often increases the robustness of analyses based on it compared with using the normal distribution.

It's a reason why it often used with risk simulations.

Comment (IvanUkhov) [inline src/distribution/logistic.rs]: OK, let’s keep it!

**Comment by DenisKolodin** [comment]
I haven't proven reason to implement default, but I've done it because R does it:

```
Logistic                 package:stats                 R Documentation

The Logistic Distribution

Description:

     Density, distribution function, quantile function and random
     generation for the logistic distribution with parameters
     ‘location’ and ‘scale’.

Usage:

     dlogis(x, location = 0, scale = 1, log = FALSE)
     plogis(q, location = 0, scale = 1, lower.tail = TRUE, log.p = FALSE)
     qlogis(p, location = 0, scale = 1, lower.tail = TRUE, log.p = FALSE)
     rlogis(n, location = 0, scale = 1)

Details:

     If ‘location’ or ‘scale’ are omitted, they assume the default
     values of ‘0’ and ‘1’ respectively.
```

I don't know the reason why it's exists. I guess that's because [some use this distribution as gaussian replacement](http://visionlab.harvard.edu/Members/Anne/Math/Logistic_vs_Gaussian.html) and defaults are similar.

Do you think we have to remove it?

File: src/distribution/logistic.rs (hunk 2 above)
Comment (DenisKolodin) [inline src/distribution/logistic.rs]: Good note! I'll repair it )

**Comment by IvanUkhov** [comment]
It looks reasonable. It’s keep the default constructor.

**Comment by DenisKolodin** [comment]
Ok, I'll keep it. Decimals repaired.

**Comment by IvanUkhov** [comment]
Thank you!

**Pull request merged**